extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectionBase.java

```diff
@@ -54,14 +58,17 @@ public String pathParam(String name) {
     }
 
     public Uni<Void> sendText(String message) {
-        Uni<Void> uni = Uni.createFrom().completionStage(() -> webSocket().writeTextMessage(message).toCompletionStage());
+        Uni<Void> uni = Uni.createFrom()
+                .completionStage(() -> webSocket().writeTextMessage(message).toCompletionStage())
+                .invoke(() -> sendingInterceptor.onSend(message));
```

Comment:
Perf-wise it might be better to only call the `sendingInterceptor` if really needed. I mean `sendText()` can be used heavily and every `.invoke(() -> sendingInterceptor.onSend(message))` we save might be useful...

Reply:
this is question of complexity vs cost of extra construct like `if`, I prefer to keep it simple as I doubt you can actually measure this, but I'd do as you wish and only call it when required

Reply:
you could replace the -> by an anonymous class.

Reply:
I'll replace it with Runnable (according to @mkouba suggestion below) which in case of NOOP is just empty instance, still my understanding is that Martin prefers to avoiding it overall.

It's not a technical problem, just matter of preference.

Reply:
@mkouba I replaced all the NOOPs with a null check and I adjusted bytecode generation to only added interception when telemetry is required; hope it's better now

also I dropped lambda